pandas/io/parsers.py

```diff
@@ -2515,6 +2515,11 @@ def _rows_to_cols(self, content):
 
             msg = ('Expected %d fields in line %d, saw %d' %
                    (col_len, row_num + 1, zip_len))
+            if len(self.delimiter) > 1 and self.quoting != csv.QUOTE_NONE:
```

Comment:
@jorisvandenbossche : If `quoting=csv.QUOTE_NONE`, all quotation marks are treated as data, so that's the user's fault, not our's.  That's why the check is necessary.